src/Enzo/enzo_EnzoMethodMHDVlct.cpp

```diff
@@ -0,0 +1,1146 @@
+
+void EnzoMethodMHDVlct::compute ( Block * block) throw()
+{
+  if (block->is_leaf()) {
+    // Check that the mesh size and ghost depths are appropriate
+    check_mesh_and_ghost_size_(block);
+    
+    // declaring Groupings that track temporary fields used for scratch space
+    // left and right reconstructed primitives
+    Grouping priml_group, primr_group;
+
+    // Names of temporary fields used to store the pressure computed from the
+    // reconstructed left and right primitives
+    // Note: in the case of adiabatic fluids, pressure is a reconstructable
+    //       quantity and fields are included for it in priml_group and
+    //       primr_group. In that case, those field names are also asigned to
+    //       pressure_name_l and pressure_name_r (this is not strictly,
+    //       necessary - EnzoEOSIdeal is smart enough to copy values if the
+    //       field names don't match)
+    std::string pressure_name_l, pressure_name_r;
+
+    // Name of fields use to store interface velocity field from the Riemann
+    // Solver to compute internal energy source term. It won't be allocated
+    // unless the dual energy formalism is in use
+    std::string interface_velocity_name;
+
+    // flux fields
+    Grouping xflux_group, yflux_group, zflux_group;
+
+    // fields used to accumulate the changes to the conserved forms of the
+    // integrable quantities and passively advected scalars (i.e. at the start
+    // of the (partial) timestep, the fields are all set to zero and are used
+    // to accumulate the flux divergence and source terms). If CT is used, it
+    // won't have space to store changes in the magnetic fields.
+    Grouping dUcons_group;
+
+    // edge-centered electric fields
+    Grouping efield_group;
+
+    // Name of the temporary field to store the central E-field (can reuse it
+    // to store different components of the field at different times)
+    std::string center_efield_name;
+
+    // face-centered weight fields - Track the x/y/z upwind direction
+    Grouping weight_group;
+
+    // temp primitive group for storing values at the half time-step
+    Grouping temp_primitive_group;
+
+    // temp interface b-fileds
+    Grouping temp_bfieldi_group;
+
+    // allocate the temporary fields (as necessary) and fill the field groupings
+    allocate_temp_fields_(block, priml_group, primr_group,
+			  pressure_name_l, pressure_name_r,
+			  interface_velocity_name,
+			  xflux_group, yflux_group, zflux_group, dUcons_group,
+			  efield_group, center_efield_name, weight_group,
+			  temp_primitive_group, temp_bfieldi_group);
+
+    // conserved-form of passively advected scalars (the initial values are
+    // taken from here and converted to specific form. Intermediate and final
+    // values get written to here)
+    Grouping *conserved_passive_scalars = cello::field_descr()->groups();
+
+    // allocate constrained transport object
+    EnzoConstrainedTransport ct = EnzoConstrainedTransport();
```

Comment:
It's actually my intention to allow for generalization on this. I just didn't include it in this PR because there weren't alternative implementations to generalize with. 

In fact, in my [development branch](https://github.com/mabruzzo/enzo-e/tree/vl_hydro_group-to-map) (which I hope to submit for merging in the next week or so), I've done some refactoring so we can turn magnetic fields off altogether.

Reply:
I think it's ok to use with only one instance. For other implementations, I got a lot of mileage out of the variations in the Gardiner&Stone 2005 paper, one is sharper than the other, but unstable, as you'd expect.